build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -204,17 +206,16 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
               // Assume we will create an outputDir.
               true,
               expectedDeletes,
-            ));
-
-    var changes = graphEvents
+            ))
         .map((data) => WatchEvent(data.type, '${data.id}'))
         .map((change) => [change]);
 
     var changeProvider = daemonOptions.buildMode == BuildMode.Auto
-        ? AutoChangeProvider(changes
+        ? AutoChangeProvider(graphEvents()
             .transform(debounceBuffer(buildOptions.debounceDelay))
             .map((changeLists) => changeLists.expand((l) => l).toList()))
```

Comment:
Should this get moved to the previous chain of function calls?

Reply:
hopefully this can just go away now as you mentioned above, will play with it